packages/packagers/js/src/ScopeHoistingPackager.js

```diff
@@ -806,7 +806,15 @@ ${code}
 
     if (imported === '*' || exportSymbol === '*' || isDefaultInterop) {
       // Resolve to the namespace object if requested or this is a CJS default interop reqiure.
-      return obj;
+      if (
+        parentAsset === resolvedAsset &&
+        this.wrappedAssets.has(resolvedAsset.id)
+      ) {
+        // Directly use module.exports for wrapped assets importing themselves.
+        return 'module.exports';
```

Comment:
If the asset is wrapped, won't `obj` already be a `parcelRequire`?

Reply:
For some reason, we have this condition:
https://github.com/parcel-bundler/parcel/blob/4700f42d81b2c99d7946371b05cd25f586e4a723/packages/packagers/js/src/ScopeHoistingPackager.js#L759
which will make `isWrapped` itself false in this self-import case.

Reply:
hmm I don't remember why...